rest/src/main/scala/io/udash/rest/annotations.scala

```diff
@@ -1,49 +1,172 @@
+  *
+  * @example
+  * {{{
+  *   trait SomeRestApi {
+  *     @POST("users/create") def createUser(@Body user: User): Future[Unit]
+  *     @PATCH("users/update") def updateUser(id: String, name: String): Future[User]
+  *   }
+  *   object SomeRestApi extends RestApiCompanion[SomeRestApi]
+  * }}}
+  */
+sealed abstract class BodyMethodTag(method: HttpMethod) extends HttpMethodTag(method)
+
+/**
+  * REST method annotated as `@GET` will translate to HTTP GET request. By default, parameters of such method
+  * are translated into URL query parameters (encoded as [[QueryValue]]). Alternatively, each parameter
+  * may be annotated as [[Path]] or [[Header]] which means that it will be translated into HTTP header value
+  *
+  * @param path see [[RestMethodTag.path]]
+  */
+class GET(val path: String = null) extends HttpMethodTag(HttpMethod.GET) {
+  @rpcNamePrefix("get_", overloadedOnly = true) type Implied
+}
+
+/** See [[BodyMethodTag]] */
+class POST(val path: String = null) extends BodyMethodTag(HttpMethod.POST) {
+  @rpcNamePrefix("post_", overloadedOnly = true) type Implied
+}
+/** See [[BodyMethodTag]] */
+class PATCH(val path: String = null) extends BodyMethodTag(HttpMethod.PATCH) {
+  @rpcNamePrefix("patch_", overloadedOnly = true) type Implied
+}
+/** See [[BodyMethodTag]] */
+class PUT(val path: String = null) extends BodyMethodTag(HttpMethod.PUT) {
+  @rpcNamePrefix("put_", overloadedOnly = true) type Implied
+}
+/** See [[BodyMethodTag]] */
+class DELETE(val path: String = null) extends BodyMethodTag(HttpMethod.DELETE) {
+  @rpcNamePrefix("delete_", overloadedOnly = true) type Implied
+}
+
+/**
+  * Causes the body parameters of a HTTP REST method to be encoded as `application/x-www-form-urlencoded`.
+  * Each parameter value itself will be first serialized to [[QueryValue]].
+  * This annotation only applies to methods which include HTTP body (i.e. not `GET`) and it must not be
+  * a method with a single body parameter ([[Body]]). Methods with single body parameter can send their body
+  * as `application/x-www-form-urlencoded` by defining custom serialization of its parameter into [[HttpBody]].
+  */
+class FormBody extends StaticAnnotation
+
+/**
+  * REST methods annotated as [[Prefix]] are expected to return another REST API trait as their result.
+  * They do not yet represent an actual HTTP request but contribute to URL path, HTTP headers and query parameters.
+  *
+  * By default, parameters of a prefix method are interpreted as URL path fragments. Their values are encoded as
+  * [[PathValue]] and appended to URL path. Alternatively, each parameter may also be explicitly annotated as
+  * [[Header]] or [[Query]].
+  *
+  * NOTE: REST method is interpreted as prefix method by default which means that there is no need to apply [[Prefix]]
+  * annotation explicitly unless you want to specify a custom path.
+  *
+  * @param path see [[RestMethodTag.path]]
+  */
+class Prefix(val path: String = null) extends RestMethodTag
```

Comment:
The docs only mention that it can be empty, not null. Is there a difference? Improve the type or docs.

Reply:
Also see: https://github.com/AVSystem/scala-commons/blob/master/docs/Annotations.md